extensions/chathistory.md

```diff
@@ -92,6 +91,18 @@ This is useful for retrieving conversation context around a single message.
 
 Request up to `limit` number of messages between the given `timestamp` or `msgid` values. With respect to the limit, the returned messages MUST be counted starting from and excluding the first message selector, while finishing on and excluding the second. This may be forwards or backwards in time.
 
+#### `TARGETS`
+
+Unlike the other subcommands, `TARGETS` does not return message history. Instead, it lists channels the user has visible history in, together with users with which the user has exchanged direct messages, ordered by the time of the latest message in the channel history or direct message conversation (i.e. sent from or to the other user). This allows the client to discover missed direct message conversations with users it is not currently aware of.
+
+    CHATHISTORY TARGETS <timestamp=YYYY-MM-DDThh:mm:ss.sssZ> <timestamp=YYYY-MM-DDThh:mm:ss.sssZ> <limit>
+
+The parameters have the same semantics as `BETWEEN`, except that they MUST be timestamps and not msgids. Returned messages have the syntax:
```

Comment:
> Returned messages have the syntax:

Have you considered using [standard replies](https://ircv3.net/specs/extensions/standard-replies)?

Reply:
Standard replies are only for errors, warnings, and such. They can't really be used for returning data for success replies. See the first sentence of the spec:

> This document specifies the standard FAIL, WARN, and NOTE messages, intended to provide a clean, consistent interface for sending general errors, warnings, and informational messages to clients.

Reply:
> The NOTE message indicates some informational message about a given command/function, or about the current session

The description for `NOTE` appears to be applicable for this case.

Reply:
NOTE is for sending an information message. There's no informational message to be sent here, only machine-readable data.

Reply:
Other commands returning lists such as LIST (which uses RPL_LIST), WHO which uses RPL_WHOREPLY use numerics.

Standard replies appears to be intended as a replacement for new numerics as far as I understand given this statement in the spec:

> Implementers should not need to reserve new numerics for error, warning, or general informational messages, especially as numerics themselves and the mapping of numerics to names can be unclear or conflicting.

Standard replies are machine readable with specified code and context. I can't think of any other defined command which returns a list with the same command name (with exception of CAP itself, which uses `*`/sender prefix after command to indicate the reply).

This proposal here does not fit into any of the existing norms. I am wondering if we avoid making yet another pattern and instead follow some existing patterns?

Reply:
I think this is mostly addressed by discussion on #357, specifically [this comment](https://github.com/ircv3/ircv3-specifications/pull/357#issuecomment-485479876)